src/chat/services/chat.service.ts

```diff
@@ -1,25 +1,47 @@
 import { ChatRepository } from '../repositories/chat.repository';
 import {
   ConflictException,
+  HttpException,
+  HttpStatus,
   Injectable,
+  Logger,
   NotFoundException,
 } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { ChatRoom } from '../schemas/chat-room.schemas';
 import * as mongoose from 'mongoose';
-import { EventsGateway } from 'src/events/events.gateway';
 import { S3Service } from 'src/common/s3/s3.service';
+import { NotificationService } from './notification.service';
+import { ChatNotification } from '../schemas/chat-notifiation.schemas';
+import { Subject, map } from 'rxjs';
 
 @Injectable()
 export class ChatService {
+  private readonly logger = new Logger(ChatService.name);
+  private readonly subject = new Subject();
   constructor(
     private readonly s3Service: S3Service,
+    private readonly notificationService: NotificationService,
     private readonly chatRepository: ChatRepository,
     @InjectModel(ChatRoom.name)
     private readonly chatRoomModel: mongoose.Model<ChatRoom>,
-    private readonly eventsGateway: EventsGateway,
+    @InjectModel(ChatNotification.name)
+    private readonly chatNotificationModel: mongoose.Model<ChatNotification>,
   ) {}
 
+  notificationListener() {
+    try {
+      return this.subject
+        .asObservable()
+        .pipe(
+          map((notification: Notification) => JSON.stringify(notification)),
+        );
+    } catch (error) {
+      this.logger.error('notificationListener : ' + error.message);
+      throw new HttpException(error.message, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
```

Comment:
```suggestion
  notificationListener() {
    return this.subject
      .asObservable()
      .pipe(
        map((notification: Notification) => JSON.stringify(notification)),
      )
      .catchError((err) => {
            this.logger.error('notificationListener : ' + error.message);
            throw new HttpException(error.message, HttpStatus.INTERNAL_SERVER_ERROR);
      })
}
```
catch 에러로 빠지는 경우가 없을거같은데 어떤 경우에 catch 로 빠지나요??
rxjs 를 활용한다면 rxjs 의 catchError을 활용하여 로직을 작성할 수도 있습니다.

Reply:
예제 코드를 따온거긴 한데 아직 어떤 종류의 error가 존재 하는지 혹은 발생 하는지 몰라서 catch문으로 그대로 냅뒀습니다! rxjs 관련 정보 감사합니다